### PR TITLE
Added types

### PR DESCRIPTION
**Pull request by TomY-Zhang**

### CODE REVIEW

**Review by Olivernyu — DISMISSED** [review]

File: server/src/types/type.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        self.fname = fname.strip().lower()
+        self.lname = lname.strip().lower()
+        self.phone = phone
+        self.posts = {}
+
+        """
+            address format: building,city,state,zip
+                i.e. 370 Jay Street,Brooklyn,NY,11201
+        """
+        address = address.lower().split()
```
Comment (leo4life2) [inline server/src/types/type.py:23]: This is fine for now, but I think later on we would want a centralized storage of all addresses in the system. This way we don't show 2 different lists when the address is off by like a whitespace or a comma.

**Review by leo4life2 — CHANGES_REQUESTED** [review]
add a gitignore file to ignore mac .DS_Store files.

File: server/src/types/type.py (hunk 1 above)
Comment (TomY-Zhang) [reply server/src/types/type.py:23]: yessir

File: server/src/types/type.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        elif type(price) != int:
+            raise ValueError('ValueError: Invalid price')
+
+        if condition is None:
+            raise ValueError('ValueError: Condition not specified')
+
+        self.title = title
+        self.details = details
+        self.condition = condition
+        self.list_date = date.today()
```
Comment (leo4life2) [inline server/src/types/type.py]: I'd say we need a parameter for `list_date` and `sold` as well. When we retrieve the data from MongoDB and reconstruct the instance we would need do directly set these fields.

**Review by leo4life2 — APPROVED** [review]
lgtm!!!

**Review by Olivernyu — APPROVED** [review]

**Pull request merged**